edit/edit.js

```diff
@@ -193,16 +194,30 @@ preinit();
           onTitleChanged: updateTitle
         });
         editor.dirty.onChange(updateDirty);
-        return Promise.resolve(editor.ready && editor.ready())
-          .then(updateDirty);
+        return editor.ready()
+          .then(updateDirty)
+          .then(!usercss && (() => {
+            // updating after init to prevent flash of validation failure on an empty name
+            $('#name').required = true;
```

Comment:
No, I think it'd be better to fix the editor to restore its initial speed that was lost during one of the storage refactors.

Reply:
So we revert this change and improve initial speed later?

Reply:
No, we move on for now because no one knows how long it'll take.

Reply:
It's probably something that you or someone familiar with tooling should kick off because one of the biggest contributors to the slow loading of our editor per Chrome's devtools profiler is that it loads a lot of small files and that alone wastes 100ms just for the network transfer. This is a ridiculously long time. A single bundle or two in a typical production page would load in a few milliseconds.

This problem is not evident in Firefox because it suppresses first paint when the page is opened. On the one hand it makes sense for many poorly built pages like ours, but on the other hand it feels much slower than Chrome which manages to show several intermediate frames [useless though] while loading.

Bundling+minifying would be also useful for our CSSLint because editing two huge files is very inconvenient. Maybe we could start with that e.g. move it to a project in `openstyles/csslint-se` where `se` is "stylus edition".

Reply:
As for why I'm against the `body` class, toggling it invalidates the layout for the entire DOM tree which requires expensive calculation. It was one of the reasons I wrote a custom keyboard-accessibility-focus implementation instead of the popular one that modified an attribute on `<html>` or `<body>`.